### PR TITLE
BasicCoin

### PR DESCRIPTION
**Pull request by jacogr**

### CODE REVIEW

File: BasicCoin.sol [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+    if (accounts[_owner].balance < _amount) throw;
+    _
+  }
+
+  modifier when_has_allowance(address _owner, address _spender, uint _amount) {
+    if (accounts[_owner].allowanceOf[_spender] < _amount) throw;
+    _
+  }
+
+  modifier when_msg_value {
```
Comment (jacogr) [inline BasicCoin.sol]: Fixed.

File: BasicCoin.sol [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+
+  address public owner = msg.sender;
+
+  function setOwner(address _new) only_owner {
+    NewOwner(owner, _new);
+    owner = _new;
+  }
+}
+
+contract BasicCoin is Owned, TokenEvents {
```
Comment (jacogr) [inline BasicCoin.sol]: Fixed.

File: BasicCoin.sol [hunk 3]
```diff
@@ -0,0 +1,123 @@
+//! BasicCoin ECR20-compliant token contract
+//! By Parity Team (Ethcore), 2016.
+//! Released under the Apache Licence 2.
+
+pragma solidity ^0.4.1;
+
+contract Token {
+  event Transfer(address indexed _from, address indexed _to, uint256 _value);
```
Comment (gavofyork) [inline BasicCoin.sol]: event parameters have no `_`

File: BasicCoin.sol [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+
+  // available token supply
+  uint public totalSupply;
+
+  // storage and mapping of all balances & allowances
+  mapping (address => Account) accounts;
+
+  // constructor sets the parameters of execution, totalSupply is in full units
+  function BasicCoin(uint128 _totalSupply) when_no_eth when_non_zero(_totalSupply) {
+    totalSupply = _totalSupply;
```
Comment (gavofyork) [inline BasicCoin.sol]: should be denoted in base units.

File: BasicCoin.sol [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+  modifier when_non_zero(uint _value) {
+    if (_value == 0) throw;
+    _;
+  }
+
+  // the base, tokens denoted in micros
+  uint constant public base = 1000000;
+
+  // available token supply
+  uint public totalSupply;
```
Comment (gavofyork) [inline BasicCoin.sol]: this should be denoted in base units.

File: BasicCoin.sol (hunk 3 above)
Comment (jacogr) [inline BasicCoin.sol]: Copied it as-is from GavCoin, however fixed here.

File: BasicCoin.sol (hunk 5 above)
Comment (jacogr) [inline BasicCoin.sol]: fixed.

File: BasicCoin.sol (hunk 4 above)
Comment (jacogr) [inline BasicCoin.sol]: fixed.

**Pull request merged**